api/database/models/deviceData.js

```diff
@@ -59,10 +60,25 @@ class DeviceDataDatabase extends DatabaseInterface {
 	 */
 	static async create(device_id, data) {
 
-		// device data collections are named by their device ID
+		// Obtain the document in the device database which corresponds to the given device_id
+		const object = await DeviceDatabase.get(device_id);
+		
+		// Create a dataRun object with the same number of dataRuns as the device of interest
+		let tempDataRun = new DataRun(object.num_dataRuns);
+		
+		// Device data collections are named by their device ID.
 		const DeviceData = await this.getCollection(device_id);
+		
+		// Determine and set the data_run (number) for the deviceData object (data)
+		tempDataRun = await tempDataRun.getDataRun(data.data.contents, DeviceData);
```

Comment:
Do you need an assignment here? tempDataRun.getDataRun looks like it modifies the tempDataRun object and then returns it. The replacement seems unnecessary.